File: src/main/java/pocket/utility/FileUtility.java
package pocket.utility;

import org.jetbrains.annotations.NotNull;

public final class FileUtility {
    @NotNull
    public static String removeExtension(@NotNull final String filepath) {
        final var lastSeparatorIndex = Math.max(filepath.lastIndexOf('/'), filepath.lastIndexOf('\\'));
        final var lastDotIndex = filepath.lastIndexOf('.');

        if (lastDotIndex > lastSeparatorIndex) {
            return filepath.substring(0, lastDotIndex);
        }

        return filepath;
    }
}

File: src/main/java/pocket/utility/FileUtility.kt
package pocket.utility

object FileUtility {
    /**
     * Removes the extension from a filepath.
     */
    fun removeExtension(filepath: String): String {
        val lastSlash = filepath.lastIndexOfAny(charArrayOf('/', '\\'))
        val lastDot = filepath.lastIndexOf('.')

        return if (lastDot > lastSlash)
            filepath.substring(0, lastDot) else filepath
    }
}